### PR TITLE
refactor: support disabling legacy diffing logic

### LINKED ISSUE #3719: Application with EnvoyFilter causes high memory/CPU usage
**Describe the bug**

An application with EnvoyFilter cause very high memory/CPU usage.

**To Reproduce**

1. Create an application with attached manifest:
[manifest.txt](https://github.com/argoproj/argo-cd/files/4741121/manifest.txt)
2. Sync application several times

**Expected behavior**

The CPU and memory usage should not spike because of single resource

**Version**

```shell
v1.5.5
```

### PR DESCRIPTION
**Pull request by alexmt**
Fixes #3719 in v1.5x branch

### CODE REVIEW

**Comment by jannfis** [comment]
* Codegen is failing due to `latest` vs `v1.5.6` in manifests, I think that's OK (manifests in 1.5 branch have 1.5 version tag, not latest which codegen would assume) - think we should support that by times
* e2e is failing with known flaky test `TestAddHelmRepoInsecureSkipVerify`, which is fixed in master but not in 1.5 release branch.

File: cmd/argocd-application-controller/main.go [hunk 1]
```diff
@@ -81,6 +82,8 @@ func newCommand() *cobra.Command {
 
 			settingsMgr := settings.NewSettingsManager(ctx, kubeClient, namespace)
 			kubectl := &kube.KubectlCmd{}
+			legacyDiffDisabled := os.Getenv("ARGOCD_POPULATE_LEGACY_DIFF") == "false"
```
Comment (jannfis) [inline cmd/argocd-application-controller/main.go:85]: Hm, I think `ARGOCD_POPULATE_LEGACY_DIFF` should be `ARGOCD_ENABKE_LEGACY_DIFF`, but that's just me.

**Review by jannfis — APPROVED** [review]
LGTM

File: cmd/argocd-application-controller/main.go (hunk 1 above)
Comment (alexmt) [reply cmd/argocd-application-controller/main.go]: I agree. `ARGOCD_ENABLE_LEGACY_DIFF` is much better.

**Pull request merged**